moe/move/move_core.py

```diff
@@ -68,7 +68,15 @@ def add_config_validator(settings: dynaconf.base.LazySettings):
 @moe.hookimpl(trylast=True)
 def edit_new_items(items: list[LibItem]):
     """Copies and formats the path of an item after it has been added to the library."""
+    # It needs to be album IDs because Album objects are not hashable.
+    album_ids_in_items = {item._id for item in items if isinstance(item, Album)}
+
     for item in items:
+        # Skip tracks and extras if their album is also being processed.
+        # This prevents double-copying since _copy_album already handles all
+        # tracks/extras.
+        if isinstance(item, (Track, Extra)) and item.album._id in album_ids_in_items:
```

Comment:
You should just be able to do something like

```python
        if not (isinstance(item, (Track, Extra)) and item.album in items):
            copy_item(item)
```

And then you don't need to deal with using the ids or the continue statement. If you had the logic reversed with the continue statement for clarity that's fine too though

Reply:
Actually I do think this is clearer! Will make the change.